conf: Align layer name to directory name

rzr opened the pull request:
Rename layer name from "genividev" to "genivi-dev"
to be used in collections.

This way this layer can be optionally and dynamicaly overloaded by sub layer,
using this scriplet:

  BBFILES += "${@' '.join('${LAYERDIR}/meta-%s/recipes*/*/*.bb' % layer \
  for layer in BBFILE_COLLECTIONS.split())}"

This is meta-ocf-automotive strategy to support specific features
for diferents distros, for more details track:
https://wiki.iotivity.org/automotive

Signed-off-by: Philippe Coval <philippe.coval@osg.samsung.com>

gunnarx commented:
Thanks @rzr. Makes sense to to have consistent naming, for any type of automation.

>This is meta-ocf-automotive strategy to support specific features
>for diferents distros, for more details track:

Do you have a more specific link and/or description of the strategy you use for including the layers?   Perhaps worthwhile to add it into the discussion I started recently on the Yocto list on best practices (or "intersting" practices at least).

By the way, you should change on wiki.iotitivity.org/automotive from: "GENIVI" to "GENIVI Development Platform, both in the text where it calls it as a Yocto based distro, and in the title.

Because "GENIVI" is not a distro, "GENIVI Development Platform is".  It's a pretty important distinction since "distro" is not the only (maybe not even the most important) part of GENIVI.

gunnarx commented:
Unfortunately the PR builds are broken.  I suspect GitHub changed something in their API, or the credentials needed to access the API (I think it does an unauthenticated anonymous access for polling changes), or some other issue - there's an error code from the plugin when trying to fetch info from GitHub.

I would expect this is OK to merge.

gunnarx commented:
It looks like we managed to break master despite evaluating that the last two merges should not affect each other (classic mistake).  So, @frznlogic let's please merge this as soon as we have a stable situation.

gunnarx commented:
Discussed on this morning's stand-up. Now that master is fixed, we just need to rebase and rebuild.

rzr commented:
ok I rebased, back to your question see at bottom of page there is an example of how distro's layers are supported in a demo layer:
https://wiki.iotivity.org/automotive

I can also share a work in progress branch as example.

gunnarx commented:
I think @frznlogic was planning to do the rebase, but you were quicker.  Thanks @rzr.

gunnarx commented:
There's still an error.  Is it because meta-erlang refers to the old name?

meta-erlang/conf/layer.conf
8:BBFILE_COLLECTIONS += "genividev-meta-rvi-append"

gunnarx commented:
We discussed today and wanted to ensure the build-checks were OK.

Then, for some reason, it seemed I had to push it again, to trigger the builds again.

But @rzr, you fixed meta-erlang right?  I think I saw that?

I'm confused now because the error is still there, but I might have goofed and pushed the wrong commit back, (because this one lacks the meta-erlang fix) ? :-(   

I tried going through reflog to find it, but I'm lost, sorry.  If you have the commit which fixes meta-erlang also, can you push it again please @rzr ? Thanks, and sorry if I was the one who goofed up here.

rzr commented:
Yes I fixed that erlang file in my previous push, Anyway I rebased and pushed again

there is not genividev anymore in this branch, but I haven't checked elsewhere.

rzr commented:
Hi,
I am unable to connect to:
https://go.genivi.org/go/pipelines/GDP-Yocto-r-car-m3-starter-kit-GITHUB-PR/150/build/1#
I assume that CI is in maintenance mode ?

frznlogic commented:
@rzr, there was some issues that's been solved now. As far as I can tell, the issues have been resolved, what do you say @gunnarx ?

rzr commented:
Should be ok now, note I an going offline until october

frznlogic commented:
Pulling this in.
